web/src/app/services/IntegrationKeyCreateDialog.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useState } from 'react'
+import { useMutation, gql } from 'urql'
+
+import { fieldErrors, nonFieldErrors } from '../util/errutil'
+
+import FormDialog from '../dialogs/FormDialog'
+import IntegrationKeyForm, { Value } from './IntegrationKeyForm'
+
+const mutation = gql`
+  mutation ($input: CreateIntegrationKeyInput!) {
+    createIntegrationKey(input: $input) {
+      id
+      name
+      type
+      href
+    }
+  }
+`
+
+export default function IntegrationKeyCreateDialog(props: {
+  serviceID: string
+  onClose: () => void
+}): JSX.Element {
+  const [value, setValue] = useState<Value | null>(null)
+  const { serviceID, onClose } = props
+
+  const [createIntegrationKeyStatus, createIntegrationKey] =
+    useMutation(mutation)
```

Comment:
just a suggestion, might read easier to shorten the names here

```suggestion
  const [createKeyStatus, createKey] =
    useMutation(mutation)
```